tests/components/laundrify/test_sensor.py

```diff
@@ -0,0 +1,106 @@
+from homeassistant.components.sensor import SensorDeviceClass
+from homeassistant.const import (
+    ATTR_DEVICE_CLASS,
+    ATTR_UNIT_OF_MEASUREMENT,
+    STATE_UNKNOWN,
+    UnitOfPower,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr
+from homeassistant.util import slugify
+from homeassistant.util.dt import utcnow
+
+from . import setup_config_entry
+
+from tests.common import async_fire_time_changed, load_fixture
+
+# Add logging to trace the flow
+logging.basicConfig(level=logging.DEBUG)
+
+
+@pytest.fixture(name="laundrify_sensor")
+def laundrify_sensor_fixture():
+    """Return a default Laundrify power sensor mock."""
+    # Load test data from machines.json
+    machine_data = json.loads(load_fixture("laundrify/machines.json"))[0]
+
+    mock_device = AsyncMock(spec=LaundrifyDevice)
+    mock_device.id = machine_data["id"]
+    mock_device.manufacturer = MANUFACTURER
+    mock_device.model = machine_data["model"]
+    mock_device.name = machine_data["name"]
+    mock_device.firmwareVersion = machine_data["firmwareVersion"]
+    return LaundrifyPowerSensor(mock_device)
+
+
+async def test_laundrify_sensor_init(
+    laundrify_sensor, hass: HomeAssistant, device_registry: dr.DeviceRegistry
+) -> None:
+    """Test Laundrify sensor default state."""
+    await setup_config_entry(hass)
+
+    sensor = laundrify_sensor
+    sensor_slug = slugify(sensor._device.name, separator="_")
+
+    state = hass.states.get(f"sensor.{sensor_slug}_power")
+    assert state.attributes[ATTR_DEVICE_CLASS] == SensorDeviceClass.POWER
+    assert state.state == STATE_UNKNOWN
+
+    device = device_registry.async_get_device({(DOMAIN, sensor._device.id)})
+    assert device is not None
+    assert device.name == sensor._device.name
+    assert device.identifiers == {(DOMAIN, sensor._device.id)}
+    assert device.manufacturer == sensor._device.manufacturer
+    assert device.model == MODELS[sensor._device.model]
+    assert device.sw_version == sensor._device.firmwareVersion
+
+
+async def test_laundrify_sensor_update(laundrify_sensor, hass: HomeAssistant) -> None:
+    """Test Laundrify sensor update."""
+    await setup_config_entry(hass)
+
+    sensor = laundrify_sensor
+    sensor_slug = slugify(sensor._device.name, separator="_")
+
+    state = hass.states.get(f"sensor.{sensor_slug}_power")
+    assert state.state == STATE_UNKNOWN
+
+    with patch("laundrify_aio.LaundrifyDevice.get_power", return_value=95):
+        future = utcnow() + timedelta(minutes=2)
+        async_fire_time_changed(hass, future)
+        await hass.async_block_till_done()
+
+        state = hass.states.get(f"sensor.{sensor_slug}_power")
+        assert state.attributes[ATTR_UNIT_OF_MEASUREMENT] == UnitOfPower.WATT
+        assert state.state == "95"
+
+
+async def test_laundrify_sensor_update_failure(
+    laundrify_sensor, hass: HomeAssistant, caplog: pytest.LogCaptureFixture
```

Comment:
Same comments as above

Reply:
Addressed in https://github.com/home-assistant/core/pull/121378/commits/6d61111b4a4f529eb108a37c18f2ebed227b1e11